### ISSUE TITLE
Components that contain props with default "() => {}" won't render description or @example

### ISSUE THREAD

**Issue by ljdz** [vue-styleguidist/vue-styleguidist#206]
A component set up like the following will correctly render the top level description and @example markdown reference:

```
<template>
</template>

<script>
/** 
 * Description.
 * @example ./examples/myComponent.md
*/
export default {
    props: {
        myObject: {
            type: Object,
            default: null
        },
    },
 }
</script>

<style scoped>
</style>
```

But the following won't render the description or the example markdown (note the only change is the default value of myObject):

```
<template>
</template>

<script>
/** 
 * Description.
 * @example ./examples/myComponent.md
*/
export default {
    props: {
        myObject: {
            type: Object,
            default: () => {}
        },
    },
 }
</script>

<style scoped>
</style>
```

Very easy to reproduce by just adding that default value "() => {}" to a prop and seeing the description and example disappear.

**Comment by ymhr** [comment]
I think you're probably getting some sort of syntax error - an arrow function in the form `() => {}` is just an arrow function with no body, it does not return an object, in fact it returns undefined. The way to get an arrow function to return an empty object would be `() => ({})`

**Comment by ljdz** [comment]
Good point, although that was not the problem -- I had also tried `function() { return {} }` without success.

However, I just noticed that if I call the function, i.e. `(function() { return {}})()` then it now works as expected.

I would call this resolved, but this now presents the issue that function props don't render correctly. I tried changing the type to `Function` and setting the default value to `function() {}`, and now the description / @example doesn't render again. Perhaps this is due to the recent fixes for `() => []` props rendering as a function in the docs?
